src/sentry/api/endpoints/organization_details.py

```diff
@@ -489,7 +491,17 @@ def put(self, request: Request, organization) -> Response:
             context={"organization": organization, "user": request.user, "request": request},
         )
         if serializer.is_valid():
-            organization, changed_data = serializer.save()
+            with transaction.atomic():
+                organization, changed_data = serializer.save()
+                if "slug" in changed_data:
+                    organization_mapping_service.create(
+                        request.user,
+                        organization.id,
+                        organization.slug,
+                        organization.stripe_id,
+                        request.data.get("idempotency_key", ""),
+                        settings.SENTRY_REGION or "us",
```

Comment:
this isn't for this PR, but it would be interesting or nice if as part of the RPC clients, mutual auth occurred and the calling region could be validated.  It seems odd that control silo need to trust the requested origin region.

Reply:
> if as part of the RPC clients, mutual auth occurred and the calling region could be validated. It seems odd that control silo need to trust the requested origin region.

:+1: To getting the region identifier from the mutual authentication credentials. With each region having separate credentials we'll be able to identify the origin region from the credentials used.